### ISSUE TITLE
How to deal with coordinate systems (CRS)?

### ISSUE THREAD

**Issue by PB-GNM** [opengeospatial/ogcapi-maps#124]
As a reaction to the call for public comment on [ogc-api-maps-part-1-core-candidate-standard ](https://www.ogc.org/requests/ogc-seeks-public-comment-on-ogc-api-maps-part-1-core-candidate-standard/), we would like to discuss on how to deal with coordinate systems (CRS).

Chapter 16, 18, 19, 20 and example B7 deal in great detail with coordinate systems. Why is the use of a (custom) CRS in this standard? In my opinion, this is a general subject that should be available as a common standard for all OGC-API’s. You could ask the same question for [OGC-API-Features part 2](https://docs.ogc.org/is/18-058r1/18-058r1.html). This standard seams to describe the CRS-subject even in more detail than the OGC-API-Features part 2, where in my opinion the subject seams more relevant for retrieving features than for retrieving maps, since features are often reused in analyses, which means that accuracy is more important than for just a map that has mostly just a viewing purpose.

Regards,
Pieter Bresters,
Geonovum

**Comment by jerstlouis** [comment]
> Why is the use of a (custom) CRS in this standard? In my opinion, this is a general subject that should be available as a common standard for all OGC-API’s. You could ask the same question for [OGC-API-Features part 2](https://docs.ogc.org/is/18-058r1/18-058r1.html).

We fully agree that dealing with CRS is a very _Common_ aspect of geospatial data access standards.
For this reason, we modeled the API building blocks in _Maps_ based on those already defined in _Features - Part 2: CRS by reference_, and this is also very closely aligned with the draft OGC API - Coverages.

While we hope that these CRS aspects end up somewhere under the umbrella of the Common SWG and/or in the [Buliding Blocks registry](https://blocks.ogc.org/), it remains that the CRS query parameters building blocks are mostly related to  resources already defined in other standards (e.g., /map and /items), where the requirements likely need to be tailored specifically for these resources.

The schema properties on the other hand mostly relate to the `/collections/{collectionId}` resource, so perhaps we could consider whether these aspects should be defined in a CRS requirement class of Common - Part 2, or a future Common extension.

At any rate, the [directive 51](https://portal.ogc.org/public_ogc/directives/directives.php#51) allows for things to be defined in a standard and then duplicated into Common. Since most of the OGC APIs want to avoid major revisions, this unfortunately may lead to forever keeping a duplicate Common version and the potential for divergence, but that is a different discussion for Common and the OAB  :)

> This standard seams to describe the CRS-subject even in more detail than the OGC-API-Features part 2, where in my opinion the subject seams more relevant for retrieving features than for retrieving maps, since features are often reused in analyses, which means that accuracy is more important than for just a map that has mostly just a viewing purpose.

There are a few reasons why we feel the CRS capability is even more important for Maps, and why we decided to include it as an optional requirement class within Part 1: Core:
- In addition to server-side map rendering, OGC API - Maps is also for accessing / subsetting / scaling pre-rendered maps or natural color imagery. Access to imagery data or base maps, on which features may be overlaid, may still have a requirement for high accuracy where a particular CRS should be used.
- Unlike Features which defaults to CRS84, the default output CRS for Maps is the native CRS (`storageCRS`), which is not necessarily CRS84. Therefore without the CRS extension, it would not be possible to retrieve a map in CRS84. This capability to re-project to CRS84 is another use case.
- Section 19 (collection map) and 20 (dataset map) mainly deal with CRS in terms of declaring their native CRS (`storageCRS`) and informing the client in response headers as to the output CRS, which is more important in Maps than in Features because there is no assumption that the output will be CRS84 by default (unlike Features and GeoJSON which assumes a CRS84 default).
- For server-side rendering, implementations may often support the use a projection library like PROJ for rendering maps in a large variety of CRSes. This ability to view things in any supported CRS is another use case. This is particularly relevant for maps displaying a large area of the world, where different projections will preserve different desirable characteristics or aesthetics.
- Compared to Features Part 2, Maps also provides support for defining custom CRS to preserve functionality which was available in WMS 1.3 "auto" crs, while providing greater flexibility to the implementation as to which projection operation methods and parameters they support, based for example on the capabilities of their projection library. This goes beyond the _by reference_ which is within the scope of Features - Part 2, providing a relatively simple mechanism to define custom CRS without requiring the complexity of parsing WKT-CRS, GML-CRS or PROJJSON, and potentially could be a Common building block that could also be supported by Features implementations.
- Though not mentioned in the comment, section 17 "orientation" provides the capability to rotate the final output as an additional transformation step conceptually applied to the output, without being part of the CRS itself. This capability is specific to visualization.
